lib/core/src/persist/chain.rs

```diff
@@ -279,22 +279,27 @@ impl Persister {
             matches!(swap.direction, Direction::Incoming),
             PaymentError::Generic {
                 err: format!(
-                    "Can only update payer_amount_sat for incoming chain swaps. Swap ID: {swap_id}"
+                    "Only an incoming chain swap can be a zero-amount swap. Swap ID: {swap_id}"
                 )
             }
         );
 
-        log::info!("Updating chain swap {swap_id}: payer_amount_sat = {payer_amount_sat}");
+        let payer_amount_sat = server_lockup_amount_sat;
```

Comment:
I assume this `server_lockup_amount_sat` is what Boltz calculates after deducting their service and server fee, so don't we need to calculate the `payer_amount_sat` including those fees?

Reply:
> I assume this `server_lockup_amount_sat` is what Boltz calculates after deducting their service and server fee

Yes.

> so don't we need to calculate the `payer_amount_sat` including those fees?

Those fees are not exposed at this point. The feerate is only available before creation, based on the chosen fees pair. Therefore, it seems to me we can only estimate the payer amount at this point in the zero-amount swap.

Reply:
Done, `payer_amount_sat` is now calculated (fetched as the lockup chain balance) and persisted:

https://github.com/breez/breez-sdk-liquid/blob/618879d551b385f5d70b331725e217af196201c7/lib/core/src/chain_swap.rs#L517-L524